go/vt/tabletserver/query_executor.go

```diff
@@ -106,8 +106,46 @@ func (qre *QueryExecutor) Execute() (reply *mproto.QueryResult) {
 			conn := qre.getConn(qre.qe.connPool)
 			defer conn.Recycle()
 			reply = qre.execSQL(conn, qre.query, true)
+		}
+		if reply != nil {
+			return reply
+		}
+		// autocommit for single dml
+		transactionID := qre.qe.txPool.Begin(qre.ctx)
+		qre.logStats.AddRewrittenSql("begin", time.Now())
+		isQueryFailed := false
+		defer func() {
+			if !isQueryFailed {
+				qre.qe.Commit(qre.ctx, qre.logStats, transactionID)
+				qre.logStats.AddRewrittenSql("commit", time.Now())
+			}
```

Comment:
That's right, but I am not sure about two things,
1. It seems exec* funcs return *mproto.QueryResult which does not have a field that indicate execution failures; therefore, it's hard to based on "reply" to find if exec failed or not.
2. It also seems exec* funcs will panic if there are errors; however, I have a feeling that call recover() and then re-panic in defer function is dangerous; and if we do not call recover, I am not how do we know there is an error. 

I feel if panic is an indication that an exec func fail, it might be better to ask exec* funcs return an error.